apps/evalite-ui/app/routes/eval.$name.lazy.tsx

```diff
@@ -0,0 +1,377 @@
+    }
+  } else {
+    evaluationWithoutLayoutShift = possiblyRunningEvaluation;
+  }
+
+  const serverState = useContext(TestServerStateContext);
+
+  const isResultRoute = useMatches({
+    select: (matches) => matches.some((m) => m.routeId.includes("result")),
+  });
+
+  const showExpectedColumn =
+    evaluationWithoutLayoutShift?.results.every(
+      (result) => result.expected !== null
+    ) ?? false;
+
+  const evalScore = average(possiblyRunningEvaluation.results || [], (r) =>
+    average(r.scores, (s) => s.score)
+  );
+
+  const prevScore = prevEvaluation
+    ? average(prevEvaluation.results, (r) => average(r.scores, (s) => s.score))
+    : undefined;
+
+  const isRunningEval =
+    serverState.isRunningEvalName(name) &&
+    evaluationWithoutLayoutShift?.created_at === mostRecentDate;
+
+  return (
+    <>
+      <title>{`${name} | Evalite`}</title>
+      <meta name="description" content={`Welcome to Evalite!`} />
+      <InnerPageLayout
+        vscodeUrl={`vscode://file${possiblyRunningEvaluation.filepath}`}
+        filepath={
+          possiblyRunningEvaluation.filepath.split(/(\/|\\)/).slice(-1)[0]!
+        }
+      >
+        <div className="text-gray-600 mb-10 text-sm">
+          <h1 className="tracking-tight text-2xl mb-2 font-medium text-gray-700">
+            {name}
+          </h1>
+          <div className="flex items-center">
+            <Score
+              evalStatus={possiblyRunningEvaluation.status}
+              isRunning={isRunningEval}
+              score={evalScore}
+              state={getScoreState(evalScore, prevScore)}
+              resultStatus={undefined}
+            ></Score>
+            <Separator orientation="vertical" className="h-4 mx-4" />
+            <span>{formatTime(possiblyRunningEvaluation.duration)}</span>
+            <Separator orientation="vertical" className="h-4 mx-4" />
+            <div className="flex items-center space-x-5">
+              <LiveDate
+                date={possiblyRunningEvaluation.created_at}
+                className="block"
+              />
+              {!isViewingLatest && (
+                <>
+                  <Link
+                    to={"/eval/$name"}
+                    params={{
+                      name,
+                    }}
+                    preload="intent"
+                    className="bg-blue-100 uppercase tracking-wide font-medium text-blue-700 px-3 text-xs py-1 -my-1 rounded"
+                  >
+                    View Latest
+                  </Link>
+                </>
+              )}
+            </div>
+          </div>
+        </div>
+
+        {history.length > 1 && (
+          <div className="mb-10">
+            <h2 className="mb-4 font-medium text-lg text-gray-600">History</h2>
+            {history.length > 1 && (
+              <MyLineChart
+                data={history}
+                onDotClick={({ date }) => {
+                  if (date === mostRecentDate) {
```

Comment:
There's an existing bug where you have to click twice on a dot if none of the scores change from the lastest run. it's not an issue with remix or tanstack router as far as I can tell and It'll need more investigation.